backend/app/controllers/spree/admin/promotions_controller.rb

```diff
@@ -28,7 +26,7 @@ def create
 
       def load_bulk_code_information
         @promotion_builder = Spree::PromotionBuilder.new(
-          base_code: @promotion.codes.first.try!(:value),
+          base_code: @promotion.base_code || @promotion.codes.first.try!(:value),
```

Comment:
If these two items are interchangeable, perhaps a migration to set that data on `base_code` would be helpful?